rules/ask-for-help/rule.md

```diff
@@ -4,14 +4,35 @@ title: Do you know when to ask for help?
 uri: ask-for-help
 authors:
   - title: Sam Wagner
+  - title: Piers Sinclair
+    url: https://ssw.com.au/people/piers-sinclair
+  - title: Brady Stroud
+    url: https://www.ssw.com.au/people/brady-stroud
 related:
   - do-you-use-timeboxing-to-avoid-wasted-time
+  - do-you-manage-up
+  - do-you-add-context-reasoning-to-your-emails
 created: 2022-08-30T04:06:01.553Z
 guid: 113b1e11-6df4-4901-9e9f-bcc06b6fb352
 ---
 When you're working, you will encounter issues that block you from progressing, force you to do a lot of research or otherwise spend time building towards being able to complete your work. In these moments, the question arises, "When should I ask for help?"
 
+Before asking for help, make sure you've done your due-diligence and exhausted your normal avenues for unblocking yourself. This concept is critical when trying to talk with Senior developers because their time is much more valuable. If you [call them without having done any research, it will waste a lot of their time and yours](https://www.ssw.com.au/rules/investigate-your-question-for-two-minutes-before-asking-someone-on-im) and won't be a good [ROI](https://www.ssw.com.au/rules/return-on-investment). Here are some examples for developers:
+
+- Try Googling it
+- Check [Stack Overflow](https://stackoverflow.com/)
+- Read the documentation e.g. [Microsoft Docs](https://docs.microsoft.com/en-au/)
+- Explore the code e.g.
+  - look at if any other pages do similar things
+  - put breakpoints and check the values to see if you can figure out little bits
+  - Systematically check things, check a tiny part, confirm your understanding, then move to the next part
+- Try to resolve it yourself a few times (at least 30 mins)
+
+
+
 As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, here's a general rule to follow:
 
 1. Find the smallest 'significant' amount of work that your project can allocate. In Scrum, this would be the smallest effort value (Extra-Small if using shirt sizes)
 2. If you spend that amount of time trying to work out the problem, you **should** ask for help
```

Comment:
@SamBWagner @pierssinclairssw 

I preferred the suggestion of asking for help once you exceed the estimate. 

```suggestion
As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, as a general rule, if you are blocked for more than the estimated time on the PBI, ask for help.
```

Reply:
I personally don't prefer it as I believe that people should ask well before exceeding a timebox in more cases than not.